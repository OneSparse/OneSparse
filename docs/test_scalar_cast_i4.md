# Scalar Casting

This documentation is also tests for the code, the examples below
show the literal output of these statements from Postgres.

Some setup to make sure warnings are shown, and that the extension
is installed.
```
set client_min_messages = 'WARNING';
create extension if not exists onesparse;

```
These casting functions cast the Postgres type integer to the
GraphBLAS scalar type GrB_INT32.
```

select cast(-1::integer as scalar);
select cast(0::integer as scalar);
select cast(1::integer as scalar);

```
These tests cast back from the scalar type GrB_INT32 to the
Postgres type integer
```

select cast(-1::integer::scalar as integer);
select cast(0::integer::scalar as  integer);
select cast(1::integer::scalar as scalar);
```