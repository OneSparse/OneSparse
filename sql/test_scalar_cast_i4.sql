set client_min_messages = 'WARNING';
create extension if not exists onesparse;
set search_path = 'onesparse';
select cast(-1::integer as scalar);
select cast(0::integer as scalar);
select cast(1::integer as scalar);
select cast(-1::integer::scalar as integer);
select cast(0::integer::scalar as  integer);
select cast(1::integer::scalar as scalar);
