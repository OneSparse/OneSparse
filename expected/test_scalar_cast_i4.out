-- # Scalar `integer` Casting
--
-- This documentation is also tests for the code, the examples below
-- show the literal output of these statements from Postgres.
--
-- Some setup to make sure warnings are shown, and that the extension
-- is installed.
set client_min_messages = 'WARNING';
create extension if not exists onesparse;
-- These casting functions cast the Postgres type `integer` to the
-- GraphBLAS scalar type `GrB_INT32`.
select cast(-1::integer as scalar);
 scalar 
--------
 i4:-1
(1 row)

select cast(0::integer as scalar);
 scalar 
--------
 i4:0
(1 row)

select cast(1::integer as scalar);
 scalar 
--------
 i4:1
(1 row)

-- These tests cast back from the scalar type `GrB_INT32` to the
-- Postgres type `integer`
select cast(-1::integer::scalar as integer);
ERROR:  operator does not exist: - scalar
LINE 1: select cast(-1::integer::scalar as integer);
                    ^
HINT:  No operator matches the given name and argument type. You might need to add an explicit type cast.
select cast(0::integer::scalar as  integer);
 int4 
------
    0
(1 row)

select cast(1::integer::scalar as scalar);
 scalar 
--------
 i4:1
(1 row)

